infrastructure/code/code.go

```diff
@@ -302,33 +300,49 @@ func (sc *Scanner) UploadAndAnalyze(ctx context.Context,
 	return issues, err
 }
 
-func (sc *Scanner) UploadAndAnalyzeWithIgnores(
-	ctx context.Context,
+func (sc *Scanner) UploadAndAnalyzeWithIgnores(ctx context.Context,
 	path string,
+	files <-chan string,
+	scanStartTime time.Time,
+	changedFiles map[string]bool,
 ) (issues []snyk.Issue, err error) {
-	response, err := codeClient.UploadAndAnalyze()
+	if scanStartTime.IsZero() {
+		scanStartTime = time.Now()
+	}
+
+	scanMetrics := codeClientObservability.NewScanMetrics(scanStartTime, 0)
+	c := config.CurrentConfig()
```

Comment:
while we are doing new signatures anyway, I'd inject the config into the function. This is something that would beneficial everywhere (but is a lot of work):
- easier to configure tests
- better for parallel processing as we don't work with global state, instead of session state

Reply:
Instead of injecting in the function I ended up refactoring the code in https://github.com/snyk/code-client-go/pull/16 because we were passing down the host too much. This was inspired by the work done in https://github.com/snyk/code-client-go/pull/14